#  Hola mundo soy pol

In [1]:
print('hola mundo soy pol Monsalvo')

hola mundo soy pol Monsalvo


In [2]:
print('Mi nombre es Pablo y tengo 44 años y me encanta la ciencia de datos.')


Mi nombre es Pablo y tengo 44 años y me encanta la ciencia de datos.


In [ ]:
85

85

# Manipulando los datos
Vamos a aprender sobre las variables en Python, cómo declararlas y utilizarlas además de conocer otros comandos de este lenguaje. :D

# Variables

In [ ]:
7

7

In [ ]:
id(7)

10751048

In [ ]:
edad = 7

In [ ]:


print(id(edad))
print(id(7))

10751048
10751048


In [ ]:
numerosiete = 7
id(numerosiete)

10751048

In [ ]:
nombre = 'Pablo'
id('Pablo')


134374334023792

In [ ]:
id(nombre)

134374334023792

# Existen algunos tipos de nombres que no podemos definir para las variables:

* Nombres que comiencen con números
* Ejemplos: 10_notas, 2_nombre_casa, etc.
* Palabras separadas por espacio
* Ejemplos: Nombre escuela, notas estudiantes, etc.
* Nombres de funciones de Python
* Ejemplos: print, type, etc.

In [ ]:
edad = 1
Edad = 2
EDAD = 3
_edad = 4
_edad_ = 5
print(edad, Edad, EDAD, _edad, _edad_)


1 2 3 4 5


In [ ]:
edad = 7
type(edad)

int

In [ ]:
promedio = 9.7
type(promedio)

float

In [ ]:
nombre = "Pablo"
type(nombre)

str

In [ ]:

v = True
type(v)

bool

En un conjunto de datos escolares podemos tener varios tipos de informaciones. Supongamos que tenemos acceso a la ficha de datos del alumno Pablo Monsalvo, cómo transformamos este nombre en variables de Python?

Ficha:
* Nombre: Pablo Monsalvo
* Edad: 44 años
* Promedio del semestre: 9,75
* Situación de aprobación: Verdadera (aprobado)

In [ ]:
nombre_estudiante = 'Pablo Monsalvo'
edad_estudiante = 44
promedio_estudiante = 9.75
aprobado = True

print(nombre_estudiante,edad_estudiante,promedio_estudiante,aprobado)
print(type(nombre_estudiante),type(edad_estudiante),type(promedio_estudiante),type(aprobado))

Pablo Monsalvo 44 9.75 True
<class 'str'> <class 'int'> <class 'float'> <class 'bool'>


# Variables numéricas

Entre los tipos de datos numéricos vamos a enfocarnos en los tipos entero y float.

Tenemos una tabla de información sobre los diversos cargos, cantidad de personas empleadas y el salario correspondiente:

* Cargo	     Cantidad  	Salario
* Vigilante	    5	       300
* Docente	      16	     500
* Coordinador	  2	       600
* Necesitamos trabajar con estos datos para obtener:

* La cantidad total de empleados;
* La diferencia entre el salario más bajo y más alto; y
* El promedio ponderado de los salarios de los empleados de la escuela.


In [ ]:

c_vigilante = 5
s_vigilante = 300

c_docente = 16
s_docente = 500

c_coordinador = 2
s_coordinador = 600

In [ ]:

total_empleados = c_vigilante + c_docente + c_coordinador
total_empleados

23

In [ ]:
diferencia_salario = s_coordinador - s_vigilante
diferencia_salario


300

In [ ]:
type(diferencia_salario)

int

In [ ]:
promedio_salarios = (c_vigilante * s_vigilante + c_docente * s_docente
                     + c_coordinador * s_coordinador) / total_empleados
promedio_salarios

465.2173913043478

In [ ]:
type(promedio_salarios)

float

# Strings
Strings hace referencia a un conjunto de caracteres formando un texto.

Podemos crear Strings cuando atribuimos un dato a una variable entre comillas sencillas (') o comillas dobles (").

In [ ]:
t = 'Alura'
type(t)


str

Las variables textuales son objetos que poseen métodos que nos ayudan a formatear strings. Los métodos los podemos ejecutar al definir un objeto de acuerdo con la siguiente estructura:

# objeto.metodo()
Existen métodos que no necesitan los (), siempre es una buena práctica verificar la documentación para cada caso.

# Situación:

Recibimos una variable con el nombre de un profesor de la escuela para añadirla a los registros. Sin embargo, necesitamos darle un tratamiento a este texto antes de insertarlo al sistema.

In [ ]:
texto = ' Pablo Matías Monsalvo '
id(texto)

132467538970864

El objetivo final es que el nombre aparezca de la siguiente forma:

'PABLO MATIAS MONSLVO'

# Convierte una string a mayúsculas.

In [ ]:
texto.upper()

' PABLO MATÍAS MONSALVO '

# Convierte una string a minúsculas.

In [ ]:
texto.lower()

' pablo matías monsalvo '

# Elimina los espacios en blanco al inicio y al final de una string.

In [ ]:
texto.strip()

'Pablo Matías Monsalvo'

# Sustituye todas las ocurrencias del texto "antiguo" en la string por el "nuevo".

In [ ]:
texto.replace('s','z')

' Pablo Matíaz Monzalvo '

# Observaciones
Los métodos devuelven una tranformación, no son ejecutados directamente sobre la variable donde está almacenado el texto.

Adicionalmente, podemos acumular la ejecución de los métodos.

In [ ]:
texto

' Pablo Matías Monsalvo '

In [ ]:
id(texto)

132467538970864

# Para que la transformación se pueda ejecutar debemos atribuir la salida de las transformaciones a las variables

In [ ]:
nuevo_texto = texto.strip().replace('s','z').upper()
print(nuevo_texto)
print(id(texto), id(nuevo_texto))

PABLO MATÍAZ MONZALVO
132467538970864 132466718546544


In [ ]:
texto = texto.strip().replace('s','z').upper()
print(texto)
print(id(texto), id(nuevo_texto))

PABLO MATÍAZ MONZALVO
132466718546832 132466718546544


# Capturando datos
En algunas aplicaciones necesitamos capturar los valores del usuario de nuestro proyecto. En python, logramos capturar los datos del usuario a través del comando input().

Para ejecutar la captura debemos atribuir el resultado de esta función a una variable.

In [ ]:
nombre = input("Escribe tu nombre: ")

Escribe tu nombre: polito


In [ ]:
type(nombre)

str

La salida de este comando siempre será una string. Esto quiere decir que aunque capturemos un valor de variable numérica, este será de tipo string.

Entonces, será necesario convertir el resultado cuando no se desea utilizar el valor como tipo string.

Existen funciones para la conversión de valores:

* Enteros: int(dato_para_conversion)
* Coma flotante: float(dato_para_conversion)
* String: str(dato_para_conversion)
* Booleano: bool(dato_para_conversion)

In [ ]:

año_admision = int(input('Digita el año de admisión: '))

Digita el año de admisión: 2025


In [ ]:
type(año_admision)


int

In [ ]:
nota_admision = float(input('Digita la nota de admisión: '))


Digita la nota de admisión: 9.9


In [ ]:
type(nota_admision)

float

Trataremos de representar mejor ahora el resultado de la transformación que obtuvimos. Nosotros podemos formatear y presentar nuestro resultado mezclando strings con valores no textuales.

Para ello utilizamos la estructura de formatación f con strings.

In [ ]:
print(f'\tEl año de admisión fue: {año_admision} \n\tLa nota de admisión fue: {nota_admision}')


	El año de admisión fue: 2025 
	La nota de admisión fue: 9.9


# Estructuras condicionales

In [ ]:
# Nota de aprobación entre 7 y 10 == nota de aprobación >= 7
# Nota de reprobación es menor a 7 == nota de reprobación < 7

"""
Si la nota es mayor o igual a 7 entonces el estudiante aprueba
Si no, entonces el estudiante reprueba
"""

'\nSi la nota es mayor o igual a 7 entonces el estudiante aprueba\nSi no, entonces el estudiante reprueba\n'

# IF y ELSE

Los comandos if y else son dos estructuras condicionales. El comando if ejecutará el bloque de comandos en caso de que la condición citada sea verdadera. El comando else ejecutará el bloque de comandos en el caso de que la condicional de if sea falsa.

El comando if es una palabra clave en Python que significa "si". Este se emplea para conformar la estructura condicional, que te permite verificar si una determinada condición es verdadera o falsa e, inmediatamente después, ejecute un bloque de código específico dependiendo del resultado de la verificación. La sintaxis para usar el comando if es:

if condicion:
    # Realiza algo

In [ ]:
if 2 < 7:
  print('La condición es verdadera.')

La condición es verdadera.


In [ ]:
if 2 > 7:
  print('La condición es verdadera.')
print('Fin de la consulta.')


Fin de la consulta.


Ya el comando else en Python se usa en junción con la palabra clave if para formar una estructura condicional. la sintaxis para usar el else es:

if condicion:
  # codigo en caso de que sea verdad
else:
  # codigo en caso de que sea falso
el comando else se ejecuta cuando la condición verificada por el if es evaluada como False.

In [ ]:
if 2 > 7:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')
print('Fin de la consulta.')

La condición es falsa.
Fin de la consulta.


# Situación:

Recibiremos el promedio de la nota de los estudiantes y necesitamos de un algoritmo que ejecute el análisis y determine si el estudiante fue Aprobado o Reprobado, mostrando un mensaje del resultado. Para ser aprobado, el promedio necesita ser igual o superior a 7.0.

In [ ]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
else:
  print('Reprobó.')

Digita la nota: 5.5
Reprobó.


Ahora, nuestra institución educativa estableció que las personas que tengan el promedio entre 5.0 y 7.0 pueden participar del curso de Recuperación durante las vacaciones para lograr aprobar.

Entonces podemos apoyarnos en un conjunto de ifs para poder estructurar esta nueva condición.

In [ ]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
if 7 > nota >= 5:
  print('Recuperación.')
if nota < 5:
  print('Reprobó.')


Digita la nota: 9
Aprobó.


Observa que en casos con 3 situaciones como este necesitamos definir bien nuestras condiciones. Pues, si realizamos una condición con else al final, este irá a considerar solamente la condición if inmediatamente anterior para generar la salida en caso de que el resultado de su operación lógica sea falso dando como resultado, dos o más ejecuciones.

Por ejemplo:

In [ ]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
if 7 > nota >= 5:
  print('Recuperación.')
else:
  print('Reprobó.')

Digita la nota: 9
Aprobó.
Reprobó.


# ELIF
El comando elif es una palabra clave en Python que significa "si no, si" y lo podemos considerar una contracción entre else e if. Se utiliza en conjunto con la palabra clave if para formar una estructura condicional en cadena.

la sintaxis para utilizar el comando elif es:

if condicion1:
    # Realiza algo
elif condicion2:
    # Realiza otra cosa
elif condicion3:
    # Realiza otra cosa
else:
    # Realiza algo diferente

El comando elif permite que puedas verificar varias condiciones en cadena, economizando espacio en tu código. Si la primera condición se evalúa como False, el interpretador de Python evaluará la próxima condición con el elif. Ello continuará hasta que una condición sea evaluada como True o hasta que el else sea alcazado. Si ninguna de las condiciones es evaluada como True, la ejecución del código del comando else será iniciada.

Vamos a emplear el mismo caso anterior:

In [ ]:

nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
elif 7 > nota >= 5:
  print('Recuperación.')
else:
  print('Reprobó.')

Digita la nota: 8.9
Aprobó.


# Operadores
Durante la construcción de comandos a veces necesitamos de una elaboración mayor de la expresión condicional, necesitando que algunos operadores lógicos se encuentren integrados.

# AND, OR, NOT
Los operadores lógicos and, or y not son usados para combinar expresiones lógicas en Python. Ellos son usados frecuentemente en conjunto con el comando if para crear estructuras condicionales más complejas.

* AND es usado para verificar si dos condiciones son verdaderas. La expresión lógica¹ x and y se evalúa como True tan solo si ambas condiciones x y y son verdaderas, y como False en caso contrario.

* OR es usado para verificar si al menos una de las condiciones es verdadera. La expresión lógica x or y se evalúa como True si al menos una de las condiciones x o y es verdadera, y como False si ambas condiciones son falsas.

* NOT es usado para negar una condición. La expresión lógica not x es evaluada como True si la condición x es falsa, y como False si la condición x es verdadera.

¹ Una expresión lógica es una declaración que puede ser evaluada como verdadera o falsa. Ella se compone por operandos lógicos² y por operadores lógicos³, que son usados ​​para combinar varias expresiones lógicas en una única expresión.

² Los operandos lógicos son los elementos que son comparados o evaluados en una expresión lógica. Ellos son generalmente valores verdaderos o falsos, pero también pueden ser expresiones lógicas más complejas. En Python, los operandos lógicos son los valores True y False.

³ Los operadores lógicos son ls símbolos o palabras clave que son usados ​​para combinar varias expresiones lógicas en una única expresión. En Python, los operadores lógicos son and, or y not, bien como las palabras clave if, elif e else.

Operadores lógicos más comunes

a	b	AND	OR	NOT
|0|0|0|0|1 |0|1|0|1|- |1|0|0|1|- |1|1|1|1|0

In [ ]:

v = True
f = False

In [ ]:
if f and f:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')

La condición es falsa.


In [ ]:
if v or v:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')


La condición es verdadera.


In [ ]:

if not f:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')


La condición es verdadera.


# IN
Es usado para verificar si un elemento está presente en una lista, tupla u otra colección de conjunto. La expresión x in y se evalúa como True si el elemento x esta presente  la variable de conjunto y, y como False en caso contrario.

Podemos verificarlo con variables de texto.



# Situación:

En la escuela nos entregaron una lista con nombres de estudiantes que fueron aprobados según el promedio del semestre, pero es necesario verificar si algunos nombres están en esa lista para confirmar que los datos estén correctos.

La lista distribuida puede ser observada a continuación:

lista = 'Juan Pérez, María González, Pedro Rodríguez, Ana López, Carlos Martínez,
    Laura Sánchez, José García, Elena Fernández, Luis Morales, Carmen Torres,
    David Ruiz, Isabel Ramírez, Javier Díaz, Sara Herrera, Miguel Castro,
    Patricia Ortega, Francisco Vargas, Marta Jiménez, Manuel Medina, Rosa Molina,
    Alejandro Silva, Silvia Ruiz, Andrés Torres, Natalia Soto, Diego Guerrero,
    Paula Ríos, Ricardo Navarro, Alicia Cordero, Carlos Vidal, Lorena Gómez'
Los nombres que necesitas verificar son los siguientes:

nombre_1 = 'Miguel Castro'
nombre_2 = 'Marcelo Noguera

In [ ]:
lista = 'Juan Pérez, María González, Pedro Rodríguez,Ana López, Carlos Martínez,Laura Sánchez, José García, Elena Fernández, Luis Morales, Carmen Torres,David Ruiz, Isabel Ramírez, Javier Díaz, Sara Herrera, Miguel Castro,Patricia Ortega, Francisco Vargas, Marta Jiménez, Manuel Medina, Rosa Molina,Alejandro Silva, Silvia Ruiz, Andrés Torres, Natalia Soto, Diego Guerrero,Paula Ríos, Ricardo Navarro, Alicia Cordero, Carlos Vidal, Lorena Gómez'


In [ ]:
nombre_1 = 'Miguel Castro'
nombre_2 = 'Marcelo Noguera'


In [ ]:
if nombre_1 in lista:
  print('El estudiante aprobó.')
else:
  print('El estudiante reprobó.')

El estudiante aprobó.


In [ ]:

if nombre_2 in lista:
  print('El estudiante aprobó.')
else:
  print('El estudiante reprobó.')

El estudiante reprobó.


# Estructuras de repetición
Cuando tenemos que ejecutar un mismo bloque de comandos durante varias ocasiones no es nada práctico hacerlo a mano.

Situación: Nos fue solicitado obtener e imprimir el promedio de 2 notas de 3 estudantes:

In [ ]:

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 2 es: {(nota_1+nota_2)/2}')

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 3 es: {(nota_1+nota_2)/2}')


Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 9
Digita la segunda nota: 8
El promedio del estudiante 2 es: 8.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 3 es: 6.0


Ahora imaginemos una situación en que no son tan solo 3 estudiantes, sino 100 estudiantes. No sería nada interesante repetir el mismo código 100 veces, sino ejecutar el mismo código 100 veces.

¡Esta estructura la logramos construir con lazos de repetición!

# WHILE
El lazo while es una estructura de control de repetición en Python que permite ejecutar un bloque de código repetidamente mientras que una determinada condición sea verdadera. Su estructura es:

while condicion:
    # bloque de codigo
Vamos a construir un ejemplo con un contador de 1 hasta 10.

In [ ]:
contador = 1
while contador <= 10:
  print(contador)
  # contador = contador + 1
  contador += 1

1
2
3
4
5
6
7
8
9
10


Ahora vamos a colectar las notas y promedios de cada alumno dentro del lazo while. Haremos un ejemplo con 3 promedios.

In [ ]:
contador = 1
while contador <= 3:
  nota_1 = float(input('Digita la primera nota: '))
  nota_2 = float(input('Digita la segunda nota: '))
  print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')
  contador += 1

Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 1 es: 6.0
Digita la primera nota: 8
Digita la segunda nota: 9
El promedio del estudiante 1 es: 8.5


# FOR
El lazo for es un tipo de estructura de control de flujo en Python que permite iterar sobre un conjunto de elementos. Su estructura es:

for elemento in conjunto:
    # codigo a ser ejecutado para cada elemento
El lazo for itera sobre cada elemento del conjunto especificado y ejecuta el bloque de código dentro del lazo para cada elemento. Cuando el lazo llega al final del conjunto, este se interrumpe y el programa continúa la ejecución después del lazo.

El conjunto lo podemos generar con la función incorporada range(). Se trata de una función capaz de generar una secuencia de números enteros. La estructura de esta función es:

range(inicio, fin, paso)
Según la documentación, range() genera una secuencia de números enteros a partir del valor del parámetro inicio hasta el valor del parámetro fin, de acuerdo con el valor del parámetro paso. Si inicio no es especificado, el valor por defecto es 0. Si paso no es especificado, el valor por defecto es 1.

Vamos a recrear el mismo contador while con el lazo for.

In [ ]:

for n in range(1,11):
  print(n)

1
2
3
4
5
6
7
8
9
10


In [ ]:
for e in range(1,4):
  nota_1 = float(input('Digita la primera nota: '))
  nota_2 = float(input('Digita la segunda nota: '))
  print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')

Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 1 es: 6.0
Digita la primera nota: 8
Digita la segunda nota: 9
El promedio del estudiante 1 es: 8.5


# Estructuras de datos
Un conjunto de elementos es una colección de items, que son almacenados juntos de manera organizada. Algunos ejemplos de conjuntos de elementos en Python son las listas, los strings y los diccionarios.

# Listas
Las listas pueden almacenar una colección de items en orden. Ellas son delimitadas por corchetes [] y sus elementos son separados por comas.

Estas pueden almacenar cualquier tipo de items, incluyendo números, strings, objetos, otras listas, entre otros. También nos permiten almacenar items de tipos de datos diferentes juntos en una única lista.

In [ ]:
lista = ['Pablo Monsalvo',9.5,9.0,9.7,True]
lista

['Pablo Monsalvo', 9.5, 9.0, 9.7, True]

Las listas son organizadas en Python porque cada elemento de la lista tiene un índice que indica su posición al interior de la lista. Los índices comienzan en 0 van hasta el tamaño de la lista menos 1.

Tenemos entonces 5 elementos con índices que varían de 0 a 4, ordenadamente:

#             [0]           [1]   [2]   [3]    [4]
lista = ['Pablo Monsalvo', 9.5 , 9.0 , 9.7 , True]
En Python tenemos también los índices negativos que se inician en el último elemento con el valor de -1 y avanzan en el universo de los negativos hasta llegar al 1° elemento:

#             [-5]         [-4]  [-3]  [-2]   [-1]
lista = ['Pablo Monsalvo', 9.5 , 9.0 , 9.7 , True]
Logramos seleccionar separadamente cada elemento a través de sus respectivos índices. Colocando el nombre de la lista y en seguida el índice que será seleccionado.

In [ ]:
type(lista[0])


str

In [ ]:
type(lista[1])


float

In [ ]:
type(lista[-1])


bool

Una forma más dinámica de trabajar item a item en una lista es utilizando un lazo for para leer un elemento a la vez.

In [ ]:
for elemento in lista:
  print(elemento)


Pablo Monsalvo
9.5
9.0
9.7
True


Situación: La nota 9.0 de Pablo necesita ser ajustada pues ganó 1 punto en su última nota por participación en clase. Entonces es necesario realizar un cambio en el índice 2 de 9.0 a 10.0

In [ ]:
lista[2] = 10.0


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True]

También podemos calcular el promedio del estudiante a partir de los datos que tenemos.




In [ ]:
promedio = (lista[1]+lista[2]+lista[3])/3
promedio

9.733333333333333

# Manipulación de listas
Las listas son muy útiles en Python porque nos permiten almacenar y acceder a una colección de items de manera organizada y rápida. Estas también ofrecen muchos métodos útiles para manipular los itens almacenados, como adicionar, remover, clasificar y buscar elementos.

# Cantidad de elementos
Usamos la función len() para descubrir la cantidad de elementos de un conjunto.

In [ ]:

len(lista)

5

# Partición
La partición de listas por indexación en Python es una técnica muy útil para seleccionar un subconjunto de elementos de una lista. Esta se realiza usando la sintaxis lista[inicio:fin], donde inicio es el índice del primer elemento que será incluído en la partición fin es el índice del primer elemento que será excluído de la partición.

In [ ]:
lista[0:2]


['Pablo Monsalvo', 9.5]

In [ ]:
lista[:2]


['Pablo Monsalvo', 9.5]

In [ ]:
lista[3:]

[9.7, True]

In [ ]:
lista[:]

['Pablo Monsalvo', 9.5, 10.0, 9.7, True]

# Añade un elemento al final de la lista.

In [ ]:
lista.append(promedio)


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333]

# extend()
Añade varios elementos al final de la lista.

Añadiremos las notas [10.0,8.0,9.0] En la lista de Pablo Monsalvo.

In [ ]:
lista.extend([10.0,8.0,9.0])


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333, 10.0, 8.0, 9.0]

A continuación te mostraremos lo que NO SE PUEDE realizarse con append.

In [ ]:

lista.append([10.0,8.0,9.0])

In [ ]:
lista

['Pablo Monsalvo',
 9.5,
 10.0,
 9.7,
 True,
 9.733333333333333,
 10.0,
 8.0,
 9.0,
 [10.0, 8.0, 9.0]]

# Remueve un elemento específico de la lista.



In [ ]:
lista.remove([10.0, 8.0, 9.0])

In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333, 10.0, 8.0, 9.0]

# Diccionario
Los diccionarios son un tipo de estructura de datos que almacenan pares de llave-valor. Estos son delimitados por llaves {} y los pares llave-valor son separados por comas.

diccionario = {llave: valor}
La llave es un elemento único que identifica a un valor en el diccionario, mientras que el valor es el item que será almacenado para la llave. Las llaves y sus respectivos valores se pueden emplear para cualquier tipo de dato.

Los diccionarios son útiles para almacenar y acceder a los datos de manera organizada y rápida. Se trata de un tipo de conjunto de elementos en Python, pues almacenan una colección de items.

In [ ]:
diccionario = {'llave_1':1,
               'llave_2':2}

diccionario


{'llave_1': 1, 'llave_2': 2}

In [ ]:
type(diccionario)

dict

# Situación:

Vamos a crear un conjunto de datos con las informaciones de matrícula de un estudiante. Los datos son los siguientes:

matrícula: 2000168933
día de registro: 25
mes de registro: 10
grupo: 2E

In [ ]:
estudiante = {
              'matricula': 2000168933,
              'dia_registro': 25,
              'mes_registro': 10,
              'grupo': '2E'
              }

estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2E'}

In [ ]:
estudiante['matricula']

2000168933

In [ ]:

estudiante['grupo']

'2E'

Es posible sustituir los valores dentro de una llave. Por ejemplo, recibimos la información de que el grupo del estudiante que registramos cambio a '2G' y ahora necesitamos cambiar el valor de la llave 'grupo'

In [ ]:
estudiante['grupo'] = '2G'

In [ ]:
estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2G'}

También podemos añadir otros datos al diccionario. Vamos a añadir la información sobre la modalidad de estudio, nuestro estudiante estudiará inicialemente en la modalidad EAD.

Definiremos una llave llamada 'modalidad' con el valor 'EAD'.

In [ ]:
estudiante['modalidad'] = 'EAD'

In [ ]:

estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2G',
 'modalidad': 'EAD'}

In [ ]:

len(estudiante)

5

# Profundizando en diccionarios
# pop()
Remueve un item de un diccionario y lo representa en la salida.

In [ ]:
estudiante.pop('grupo')

'2G'

In [ ]:
estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'modalidad': 'EAD'}

In [ ]:
len(estudiante)

4

# items()
Retorna una lista de pares llave-valor del diccionario.

In [ ]:
estudiante.items()


dict_items([('matricula', 2000168933), ('dia_registro', 25), ('mes_registro', 10), ('modalidad', 'EAD')])

# keys()
Retorna una lista de las llaves del diccionario.

In [ ]:
estudiante.keys()

dict_keys(['matricula', 'dia_registro', 'mes_registro', 'modalidad'])

# values()
Retorna una lista de los valores del diccionario.

In [ ]:
estudiante.values()

dict_values([2000168933, 25, 10, 'EAD'])

# Lectura de valores con el lazo for

In [ ]:
for llave in estudiante.keys():
  print(llave)

matricula
dia_registro
mes_registro
modalidad


In [ ]:
for llave,valor in estudiante.items():
  print(llave,' --> ', valor)

matricula  -->  2000168933
dia_registro  -->  25
mes_registro  -->  10
modalidad  -->  EAD


# VAMOS POL!!!